### PR TITLE
Support multi-planar camera capability (Bugfix)

### PR DESCRIPTION
**Pull request by baconYao**
## Description
Some camera deivce only support the multi-planar API through the Video Capture interface, therefore, improve this script to support it.

1: https://www.kernel.org/doc/html/v5.15/userspace-api/media/v4l/dev-capture.html#querying-capabilities
2: https://www.kernel.org/doc/html/v5.15/userspace-api/media/v4l/vidioc-querycap.html?highlight=vidioc_querycap#description

## Resolved issues

- Baoshan MIPI ISP camera
  - Failed scenario:
    - submission: https://certification.canonical.com/hardware/202202-29979/submission/331376/test-results/?term=camera
    - The `/dev/video0`, `mtk-camsv-isp30`, is MIPI ISP camera which supports the `Multi-planar` only.

## Tests
- Baoshan MIPI ISP camera
  - Pass after fixing it
    - https://certification.canonical.com/hardware/202202-29979/submission/331520/test-results/?term=camera

### CODE REVIEW

File: providers/base/bin/camera_test.py [hunk 1]
```diff
@@ -193,8 +194,10 @@ def detect(self):
                 "    version: %s.%s.%s"
                 % (cp.version >> 16, (cp.version >> 8) & 0xff,
                    cp.version & 0xff))
+            capture_cability = cp.capabilities & V4L2_CAP_VIDEO_CAPTURE or \
```
Comment (pieqq) [inline providers/base/bin/camera_test.py:197]: typo: I guess you meant `capability`, but maybe it's better to stick to the plural anyway (`capture_capabilities`).

**Review by pieqq — CHANGES_REQUESTED** [review]
Good investigation, thanks for pointing to the reference documentation!

I really like the fact that your commit message contains the links to the explanation on what you are trying to achieve! There are a few typos in it, though, it would be nice to fix them.

Since we started enforcing more testing, especially on the scripts side, could you implement some [unit tests](https://github.com/canonical/checkbox/blob/main/CONTRIBUTING.md#writing-and-running-unit-tests-for-providers) for this script? At the very least, a test to check that the feature you added works (this would help with regressions in the future). Feel free to ask for help if you need.

Last thing: Can you modify the title of your PR to fit into [one of the markers](https://github.com/canonical/checkbox/blob/main/.github/pull_request_template.md?plain=1#L2-L15)? (yours would probably be "BugFix" at the end of the title). Also, there is a typo in the title itself :)

File: providers/base/bin/camera_test.py (hunk 1 above)
Comment (baconYao) [reply providers/base/bin/camera_test.py]: Fix the typo and append the Bugfix to the title.

I will land the unit test latter.

**Comment by baconYao** [comment]
Hi Checkbox Team, I've refactored `camera_test.py` and add some unit test cases for it. Please help to review it, thanks.

By the way, the following links are the sideload result to show the refactor don't break any original logic.
- Test with USB camera
  - https://certification.canonical.com/hardware/202303-31430/submission/333922/
- Test without USB camera
  - https://certification.canonical.com/hardware/202303-31430/submission/333925/

**Review by pieqq — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**